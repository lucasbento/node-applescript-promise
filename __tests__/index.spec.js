import fs from 'fs-extra';
import applescript from '../src/index';

it('should execute file returning a promise', async () => {
  const fixturePath = `${__dirname}/../fixtures/snippetWithArgument.applescript`;
  const arg = 'should print this string';

  const response = await applescript.execFile(fixturePath, [arg]);

  expect(response).toBe(arg);
});

it('should execute string returning a promise', async () => {
  const fixturePath = `${__dirname}/../fixtures/snippet.applescript`;

  const script = await fs.readFile(fixturePath, 'utf8');
  const response = await applescript.execString(script);

  expect(response).toMatchSnapshot();
});
