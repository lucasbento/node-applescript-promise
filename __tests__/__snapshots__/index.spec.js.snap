// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should execute string returning a promise 1`] = `"this is snippet without argument"`;
